src/duration.js

```diff
@@ -236,11 +237,28 @@ export default class Duration {
    * @return {Duration}
    */
   static fromObject(obj) {
-    return new Duration({
-      values: normalizeObject(obj, Duration.normalizeUnit, true),
-      loc: Locale.fromObject(obj),
-      conversionAccuracy: obj.conversionAccuracy
-    });
+    if (obj == null || typeof obj !== 'object') {
+      throw new InvalidArgumentError(
+        'Duration.fromObject(arg): argument expected to be an object.'
+      );
+    }
+    // If Argument is empty object {} then create zero Duration;
+    if (Object.getOwnPropertyNames(obj).length === 0) {
+      return new Duration({ values: {} });
+    }
+    const val = normalizeObject(obj, Duration.normalizeUnit, true);
+    const hasUnits = Object.keys(val).length > 0;
+    if (hasUnits) {
+      return new Duration({
+        values: val,
+        loc: Locale.fromObject(obj),
+        conversionAccuracy: obj.conversionAccuracy
+      });
+    } else {
+      throw new InvalidArgumentError(
```

Comment:
Is it OK? Maybe create invalid duration will be better?

Reply:
Yeah, invalid is probably better here